rmw/include/rmw/serialized_message.h

```diff
@@ -29,9 +29,49 @@ extern "C"
  * more complex and is therefore aliased.
  */
 typedef rcutils_uint8_array_t rmw_serialized_message_t;
+
+/// Return a zero initialized serialized message struct.
 #define rmw_get_zero_initialized_serialized_message rcutils_get_zero_initialized_uint8_array
+
+/// Initialize a serialized message, zero initializing its contents.
+/**
+ * \param[inout] serialized_message a pointer to the serialized message to initialize
+ * \param[in] message_capacity the size of the memory to allocate
+ * \param[in] allocator the allocator to use for the memory allocation
+ * \return `RMW_RET_OK` if successful, or
+ * \return `RMW_RET_INVALID_ARGUMENT` if any arguments are invalid, or
+ * \return 'RMW_RET_BAD_ALLOC` if no memory could be allocated correctly
+ * \return `RMW_RET_ERROR` if an unexpected error occurs
+ */
 #define rmw_serialized_message_init rcutils_uint8_array_init
+
+/// Finalize a serialized message.
+/**
+ * Passing an uninitialized instance to this function leads to undefined
+ * behavior.
+ *
+ * \param[in] serialized_message pointer to the serialized message to be cleaned up
+ * \return `RMW_RET_OK` if successful, or
+ * \return `RMW_RET_INVALID_ARGUMENT` if the serialized_message argument is invalid
+ * \return `RMW_RET_ERROR` if an unexpected error occurs
+ */
 #define rmw_serialized_message_fini rcutils_uint8_array_fini
+
+/// Resize the internal buffer of the serialized message.
+/**
+ * The internal buffer of the serialized message can be resized dynamically if needed.
+ * If the new size is smaller than the current capacity, then the memory is
+ * truncated.
+ * Be aware, that this might deallocate the memory and therefore invalidates any
+ * pointers to this storage.
```

Comment:
This sentence is grammatically correct, but I would still find a modification slightly easier to read:

```suggestion
 * Be aware, that this might deallocate the memory and therefore invalidate any
 * pointers to this storage.
```

Reply:
Sounds good. At some point we should correct `rcutils_uint8_array_t` API documentation as well.